### ISSUE TITLE
When ~denote-org-capture-specifiers~ don't include a format specificer for a link don't call ~org-get-id-create~

### ISSUE THREAD

**Issue by grszkthfr** [protesilaos/denote#404]
Hi Prot,

thanks for denote, it's a real pleasure to use!

I just setup the entry point to denote via org-capture. I do not want to store a link to the file where I started the capture from.

As such I have set `(setq denote-org-capture-specifiers "%?")`. However, the org heading I am capturing from still gets an ID property.

Is it possibile to have more sensible defaults/checks in place to skip this step when a link shouldn't be included in the denote note? In general, I would prefer not to populate the headings with IDs that are never used in the end.

Best
Jonas

**Comment by protesilaos** [comment]
> From: Jonas Großekathöfer ***@***.***>
> Date: Fri,  2 Aug 2024 01:59:29 -0700
>
> Hi Prot,

Hello Jonas,

> thanks for denote, it's a real pleasure to use!

You are welcome!

> I just setup the entry point to denote via org-capture. I do not want
> to store a link to the file where I started the capture from.
>
> As such I have set `(setq denote-org-capture-specifiers "%?")`.
> However, the org heading I am capturing from still gets an ID
> property.
>
> Is it possibile to have more sensible defaults/checks in place to skip
> this step when a link shouldn't be included in the denote note? In
> general, I would prefer not to populate the headings with IDs that are
> never used in the end.

Is that 'setq' all you have for this setup? In which case do headings
get the ID? Are we talking only about Denote files, or more generally?

I ask because I want to reproduce this. I think what you describe makes
sense and we can make it optional. But I need to see it in action to
better understand the scenario.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by grszkthfr** [comment]
Am 02.08.2024 um 13:53 schrieb Protesilaos Stavrou ***@***.***>:﻿
> From: Jonas Großekathöfer ***@***.***>

> Date: Fri,  2 Aug 2024 01:59:29 -0700

>

> Hi Prot,



Hello Jonas,



> thanks for denote, it's a real pleasure to use!



You are welcome!



> I just setup the entry point to denote via org-capture. I do not want

> to store a link to the file where I started the capture from.

>

> As such I have set `(setq denote-org-capture-specifiers "%?")`.

> However, the org heading I am capturing from still gets an ID

> property.

>

> Is it possibile to have more sensible defaults/checks in place to skip

> this step when a link shouldn't be included in the denote note? In

> general, I would prefer not to populate the headings with IDs that are

> never used in the end.



Is that 'setq' all you have for this setup? Basically, yes. You can find the org-capture relevant denote part here: https://gitlab.com/grszkthfr/dotfiles/-/blob/master/emacs/README.org?plain=1#L563-574It‘s taken from denotes documentation with a custom key („z“) and appended to the list. Note that I have a few org-id-* settings. In which case do headings get the ID? When I tried the capture template with the cursor within the „* Capture“-heading of the above linked README.org a property drawer with an ID was created. I didn‘t test with other files.Are we talking only about Denote files, or more generally?Generally. In my example above the cursor was not on a denote file.
I ask because I want to reproduce this. I think what you describe makes

sense and we can make it optional. But I need to see it in action to

better understand the scenario.Hope that helps!Best Jonas
-- 

Protesilaos Stavrou

https://protesilaos.com

**Comment by grszkthfr** [comment]
> Is that 'setq' all you have for this setup? 

Basically, yes. You can find the org-capture relevant denote part here: https://gitlab.com/grszkthfr/dotfiles/-/blob/master/emacs/README.org?plain=1#L563-574It‘s taken from denotes documentation with a custom key („z“) and appended to the list. Note that I have a few org-id-* settings. 

> In which case do headings get the ID? 

When I tried the capture template with the cursor within the „* Capture“-heading of the above linked README.org a property drawer with an ID was created. I didn‘t test with other files.

> Are we talking only about Denote files, or more generally?

Generally. In my example above the cursor was not on a denote file.

> I ask because I want to reproduce this. I think what you describe makes sense and we can make it optional. But I need to see it in action to better understand the scenario.

Hope that helps!

Best Jonas

**Comment by protesilaos** [comment]
Thank you @grszkthfr! I just pushed a change. If you can give it a try, please let me know how it works for you.

**Comment by grszkthfr** [comment]
Hi Prot,

I gave it a try and it seems that the problem is on my end and more general.

While testing your change I noticed that the ID is already created when opening the org-capture dispatcher and without calling any template... This was also true removing the code section that plugs denotes functionality into org-capture.

I'll investigate this further and report back as soon as I have some insights on your fix.

Sorry, if I have caused unnecessary fuss...
